components/core/src/clp/clp/FileDecompressor.hpp

```diff
@@ -25,13 +29,29 @@ class FileDecompressor {
             std::unordered_map<std::string, std::string>& temp_path_to_final_path
     );
 
-    bool decompress_to_ir(
+    /**
+     * Decompresses the given file split into one or multiple IR chunks.
+     * The method creates a new IR chunk when the current IR chunk exceeds ir_target_size.
+     *
+     * Note: The decompressed IR chunks are first written to the temp_output_dir.
+     * Caller must provide an ir_output_handler which will handle where and how
+     * the IR chunk shall be finally outputted.
+     * @tparam IrOutputHandler　
+     * @param archive_reader
+     * @param file_metadata_ix
+     * @param ir_target_size target size of each IR chunk
+     * @param temp_output_dir directory to store temporary IR chunks
+     * @param ir_output_handler function handling the final output
+     * @return Whether decompression was successful
+     */
```

Comment:
```suggestion
    /**
     * Decompresses the given file split into one or more IR files (chunks). The method creates a
     * new IR chunk when the current IR chunk exceeds ir_target_size.
     *
     * @tparam IrOutputHandler Method to handle IR chunks output by this method.
     * Signature: (boost::filesystem::path const& ir_file_path, string const& orig_file_id,
     * size_t begin_message_ix, size_t end_message_ix) -> bool;
     * The method returns whether it succeeded.
     * @param archive_reader
     * @param file_metadata_ix
     * @param ir_target_size Target size of each IR chunk. NOTE: This is not a hard limit.
     * @param temp_output_dir Directory to write IR chunks to
     * @param ir_output_handler
     * @return Whether decompression was successful.
     */
```

I don't think this method needs to concern itself with:
* what `ir_output_handler` does
* whether the output directory is temporary or not

So how about also  renaming `temp_output_dir` to `output_dir`?

Reply:
I would prefer to distinguish `temp_output_dir` from `output_dir`, since the temporary IR will be moved eventually so technically the directory doesn't contain "output".

It can be named as something else, but I think we should distinguish it from the real `output_dir`

Reply:
>Method to handle IR chunks output by this method. 
This reads a bit weird to me, can we say `Method to handle IR chunks output by this function`  or `Function to handle IR chunks output by this method.`

Reply:
> I would prefer to distinguish `temp_output_dir` from `output_dir`, since the temporary IR will be moved eventually so technically the directory doesn't contain "output".
> 
> It can be named as something else, but I think we should distinguish it from the real `output_dir`

This method has no knowledge of what is the real output directory though, right? In fact, in the scenario where we decompress IR files to the filesystem (rather than block storage), then the `temp_output_dir` is actually the final output directory. Also consider that if you're a new developer reading this method, it's confusing why there's a `temp_output_dir` argument but there's no `output_dir` argument.

Reply:
> > Method to handle IR chunks output by this method.
> 
> This reads a bit weird to me, can we say `Method to handle IR chunks output by this function` or `Function to handle IR chunks output by this method.`

Gotcha. Let's change all instances of `method` to `function` and let's change the param description to `Function to handle the resulting IR chunks.`

Reply:
>In fact, in the scenario where we decompress IR files to the filesystem (rather than block storage), then the temp_output_dir is actually the final output directory.

Technically this is inaccurate since the behavior is up to the IR handler (i.e. caller), but get your point. I agree with the change.